common-utils/src/common/deployment/__init__.py

```diff
@@ -179,31 +162,37 @@ class FlowDockerEnv(BaseModel):
     dockerfile_path: FilePath
     python_version: Literal["3.10"]
     docker_build_context: DirectoryPath = Path(".")
+    project_name: StrictStr
     _image_name: str = PrivateAttr()
 
     def build_image(self) -> None:
         """Build a docker image using model attributes."""
+        # get proper image name and set instance private attr
         image_name = create_image_name_str(
-            self.env_name, self.python_version  # type: ignore
+            self.project_name, self.env_name, self.python_version
         )
         self._image_name = image_name
+        # build more args and run image build
         dockerfile_path = self.dockerfile_path
         python_version = self.python_version
         docker_build_context = self.docker_build_context
         run_command(
             f"{SCRIPT_PATH}/build_image.sh {image_name} {dockerfile_path} {python_version} {docker_build_context}"
         )
 
-    def push_image(self, account_id: str) -> str:
+    def push_image(self) -> str:
         """Push built image to ECR
 
-        Args:
-            account_id (str): Account id for building ECR URL.
-
         Returns:
             str: Pushed ECR image name.
         """
+        # only need AWS SDK on deployment
+        from boto3 import session
+
+        init_session = session.Session()
+        account_id = get_aws_account_id(init_session)
```

Comment:
we do not need the boto3 library or the resulting account_id until the `push_image` function is called.  This all apart of the process of reducing complexity and global reliance on prefect-aws library.